ReactNativeClient/lib/shim-init-react.js

```diff
@@ -191,6 +192,26 @@ function shimInit() {
 		if (!(name in injectedJs)) throw new Error(`Cannot find injectedJs file (add it to "injectedJs" object): ${name}`);
 		return injectedJs[name];
 	};
+
+	shim.loadSecureItems = (Setting, items) => {
+		for (let i = 0; i < items.length; i++) {
+			const c = items[i];
+
+			c.value = Setting.formatValue(c.key, c.value);
+			c.value = Setting.filterValue(c.key, c.value);
+
+			Setting.cache_.push(c);
+		}
+		Setting.dispatchUpdateAll();
+	};
+
+	shim.saveSecureItems = async (Setting, items) => {
+		const queries = [];
+		for (let i = 0; i < items.length; i++) {
+			queries.push(Database.insertQuery(Setting.tableName(), items[i]));
+		}
+		await Setting.db().transactionExecBatch(queries);
+	};
```

Comment:
If you refactor the functions as described above you won't need all this duplicated code.

Reply:
You're right. I'll refactor the code to avoid duplication.